File: bitcode_android/Day05/app/src/main/java/in/kaligotla/day05/SecondActivity.java
package in.kaligotla.day05;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {
    TextView textView;
    String name, city;
    int rollNumber;

    Button btnPrevious;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        initViews();
        extractData();
        initListeners();
    }

    public void initViews() {
        textView = findViewById(R.id.txtView);
        btnPrevious = findViewById(R.id.btnPrevious);
    }

    public void extractData() {
        Intent i = getIntent();
        Bundle bundle = i.getExtras();
        name = bundle.getString("name");
        rollNumber = bundle.getInt("roll_number");
        city = bundle.getString("city");
        textView.setText(name+ ", "+ rollNumber + ", " + city);
    }

    public void initListeners() {
        btnPrevious.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent();
                intent.putExtra("state", "Maharashtra");
                setResult(1,intent);
//                finish();
            }
        });
    }
}

File: bitcode_android/NavigationDemo/app/src/main/java/in/kaligotla/navigationdemo/SecondActivity.kt
package `in`.kaligotla.navigationdemo

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import `in`.kaligotla.navigationdemo.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {
    private lateinit var activitySecondBinding: ActivitySecondBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activitySecondBinding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(activitySecondBinding.root)
        activitySecondBinding.previous.setOnClickListener {
            val intent: Intent = Intent()
            intent.putExtra("key","value")
            setResult(1, intent)
            finish()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        val b: Bundle = data?.extras!!
        Toast.makeText(this, b.getString("key"), Toast.LENGTH_SHORT).show()
    }
}